### PR TITLE
warnings and social thumbnails fixes

### PR DESCRIPTION
**Pull request by felpsio**

### CODE REVIEW

File: src/templates/article.js [hunk 1]
```diff
@@ -53,13 +54,14 @@ const ArticleTemplate = props => {
...
   const pathname = post.fields.slug
+
   return (
     <Layout location={props.location} title={siteTitle}>
       <SEO
         description={post.frontmatter.description || post.excerpt}
         pathname={pathname}
         title={post.frontmatter.title}
-        image={post.frontmatter.image}
+        image={post.frontmatter.image.publicURL}
```
Comment (Swizec) [inline src/templates/article.js:80]: Is this auto generated or will I have to manually specify it everywhere?

File: src/templates/article.js [hunk 2]
```diff
@@ -68,18 +70,19 @@ const ArticleTemplate = props => {
...
-            or <a href={convertkitURL}>subscribe by email 💌</a>
+            or <a href={convertkitURL}>subscribe by email <span role="img" aria-label="heart">💌</span></a>
           </em>
         </p>
         <div
           dangerouslySetInnerHTML={{
             __html: post.html,
           }}
         />
+        <Reactions pathname={props['*']} />
```
Comment (Swizec) [inline src/templates/article.js:101]: Why asterisk as the prop name for this?

File: src/widgets/About.js [hunk 3]
```diff
@@ -17,19 +17,19 @@ const WrapperGroup = styled.div`
...
-    height: 300px;
-    grid-area: Picture;
-  }
-  @media (max-width: 940px) {
-    img {
-      height: 200px;
-    }
-  }
-`
+// const WrapperImage = styled.div`
```
Comment (Swizec) [inline src/widgets/About.js:20]: Don't these styles do anything?

File: src/widgets/About.js [hunk 4]
```diff
@@ -41,28 +41,28 @@ const Features = styled.div`
...
-    grid-template-columns: 1fr;
-    grid-template-areas:
-      'Picture'
-      'Copy'
-      'Feat';
-  }*/
-  }
 `
 
+/*{
```
Comment (Swizec) [inline src/widgets/About.js:46]: Don't these styles do anything?

File: src/widgets/TableOfContents.js [hunk 5]
```diff
@@ -37,202 +38,383 @@ const Wrapper = styled.div`
...
-        <p>Will you buy a christmas tree? - a pie chart</p>
-        <p>What goes in Christmas stockings - a piechart with tooltips</p>
-        <p>When Americans buy Christmas presents - a curved line chart</p>
-        <p>When people buy candy - animated barchart with easing</p>
-        <p>A responsive stack chart of smartphone market share</p>
-        <p>A Sankey diagram</p>
-        <p>Try Uber's WebGL dataviz library</p>
-        <p>Real-time WebGL map of all airplanes in the world</p>
-        <p>A compound arc chart</p>
+        <Topic
```
Comment (Swizec) [inline src/widgets/TableOfContents.js:62]: Awesome refactor, thanks!

File: src/widgets/About.js (hunk 3 above)
Comment (felpsio) [reply src/widgets/About.js:20]: looks like they weren't used anyway

File: src/widgets/TableOfContents.js (hunk 5 above)
Comment (felpsio) [reply src/widgets/TableOfContents.js:62]: :)

File: src/widgets/About.js (hunk 4 above)
Comment (felpsio) [reply src/widgets/About.js:46]: looks like they weren't used anyway

File: src/templates/article.js (hunk 2 above)
Comment (felpsio) [reply src/templates/article.js:101]: to get the pathname. It's the same way I used on serverless handbook

File: src/templates/article.js (hunk 1 above)
Comment (felpsio) [reply src/templates/article.js:80]: autogenerated

File: gatsby-node.js [hunk 6]
```diff
@@ -75,25 +75,25 @@ exports.createPages = ({ graphql, actions }) => {
     }
 
     // create pages pages
-    {
```
Comment (Swizec) [inline gatsby-node.js:78]: Don't we need this to create article pages?

File: src/templates/article.js [hunk 7]
```diff
@@ -48,11 +50,25 @@ const Wrapper = styled.div`
...
 
 const ArticleTemplate = props => {
+  useEffect(() => {
+    if (typeof window !== 'undefined') {
+      window.requestAnimationFrame(
+        () => {
+          let nodes = document.querySelectorAll('.remark-cta')
+          
+          for (let i = 0; i < nodes.length; i++) {
+            ReactDOM.render((<CTA />), nodes.item(i))
```
Comment (Swizec) [inline src/templates/article.js]: I still don't quite understand why we can't do this during compile time.

**Review by Swizec — APPROVED** [review]

**Pull request merged**